app/controllers/merchant/items_controller.rb

```diff
@@ -1,7 +1,77 @@
-class Merchant::ItemsController < ApplicationController
+class Merchant::ItemsController < Merchant::BaseController
+  def index
+    @merchant = Merchant.find(current_user.merchant_id)
+    @items = @merchant.items
+    render 'items/index'
+  end
+
+  def new
+    @merchant = Merchant.find(current_user.merchant_id)
+    @item = Item.new
+  end
+
+  def create
+    @merchant = Merchant.find(current_user.merchant_id)
+    @item = @merchant.items.new(item_params)
+    check_default_image(@item)
+    if @item.save
+      flash[:alert] = 'New item saved successfully!'
+      redirect_to "/merchant/items"
+    else
+      flash[:error] = @item.errors.full_messages.to_sentence
+      render :new
+    end
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+  end
 
-  def index 
+  def update
+    @item = Item.find(params[:id])
+    @item.update(item_params) if params[:item]
+    check_default_image(@item) unless params[:status]
+    if params[:status]
+      update_status
+    elsif @item.save
+      flash[:alert] = "Item #{@item.id} has been successfully updated!"
+      redirect_to "/merchant/items"
+    else
+      flash[:error] = @item.errors.full_messages.to_sentence
+      render :edit
+    end
+  end
+
+  def destroy
+    item = Item.find(params[:id])
+    Review.where(item_id: item.id).destroy_all
+    item.destroy
+    flash[:alert] = 'This item is now deleted.'
+    redirect_to "/merchant/items"
+  end
 
+  def update_status
+    @item = Item.find(params[:id])
+    if params[:status] == 'deactivate'
+      @item.update(:active? => false)
+      flash[:alert] = 'This item is no longer for sale.'
+    elsif params[:status] == 'activate'
+      @item.update(:active? => true)
+      flash[:alert] = 'This item is available for sale.'
+    end
+    redirect_to "/merchant/items"
   end
 
+  private
+
+  def item_params
+    params.require(:item).permit(:name,:description,:price,:inventory,:image)
+  end
+
+  def check_default_image(item)
+    if params[:item][:image] == ''
+      item.update(image: '/images/image.png')
+      item.save
+    end
+  end
 end
```

Comment:
Regarding the duplication of this file, is it possible to put this into a module and include it within each controller? Not sure if that would be good practice as Brian didn't mention that as a possibility.

Reply:
Talking to Brian about it tomorrow!